alkira/resource_alkira_connector_gcp_interconnect_helper.go

```diff
@@ -137,71 +101,64 @@ func expandGcpInterconnectInstances(in []interface{}, so []interface{}, m interf
 	var segmentOptions []map[string]interface{}
 
 	// loop over the segment options
-	for _, cSegmentOption := range d.Get("segment_options").([]interface{}) {
-		cfgSegmentOption := cSegmentOption.(map[string]interface{})
-		for _, aSegmentOption := range instance.SegmentOptions {
-			// make segmentOptions map for each instance using the instance name
-			if cfgSegmentOption["instance_name"].(string) == instance.Name {
-				segmentId, err := getSegmentIdByName(aSegmentOption.SegmentName, m)
-				if err != nil {
-					log.Printf("[ERROR] error getting segment ID for Segment Name %v", aSegmentOption.SegmentName)
-					return nil, err
-				}
+	for _, aSegmentOption := range sO {
+		segmentId, err := getSegmentIdByName(aSegmentOption.SegmentName, m)
+		if err != nil {
+			log.Printf("[ERROR] error getting segment ID for Segment Name %v", aSegmentOption.SegmentName)
+			return nil, err
+		}
 
-				// create a list of map for customer gateways
-				var customerGateways []map[string]interface{}
-				for _, aCustomerGateway := range aSegmentOption.CustomerGateways {
-					// add all gateways to the list
-					customerGateway := map[string]interface{}{
-						"loopback_ip":  aCustomerGateway.LoopbackIp,
-						"tunnel_count": aCustomerGateway.TunnelCount,
-					}
-					customerGateways = append(customerGateways, customerGateway)
-				}
-				segmentOption := map[string]interface{}{
-					"instance_name":            instance.Name,
-					"segment_id":               segmentId,
-					"advertise_on_prem_routes": aSegmentOption.AdvertiseOnPremRoutes,
-					"disable_internet_exit":    aSegmentOption.DisableInternetExit,
-					"customer_gateways":        customerGateways,
-				}
-				segmentOptions = append(segmentOptions, segmentOption)
+		// create a list of map for customer gateways
+		var customerGateways []map[string]interface{}
+		for _, aCustomerGateway := range aSegmentOption.CustomerGateways {
+			// add all gateways to the list
+			customerGateway := map[string]interface{}{
+				"loopback_ip":  aCustomerGateway.LoopbackIp,
+				"tunnel_count": aCustomerGateway.TunnelCount,
 			}
+			customerGateways = append(customerGateways, customerGateway)
 		}
+		segmentOption := map[string]interface{}{
+			"segment_id":               segmentId,
+			"advertise_on_prem_routes": aSegmentOption.AdvertiseOnPremRoutes,
+			"advertise_default_routes": !aSegmentOption.DisableInternetExit,
+			"customer_gateways":        customerGateways,
+		}
+		segmentOptions = append(segmentOptions, segmentOption)
 	}
 	return segmentOptions, nil
 }
 
 func setGcpInterconnectInstance(d *schema.ResourceData, connector *alkira.ConnectorGcpInterconnect, m interface{}) {
 	var instances []map[string]interface{}
-	var segmentOptions []map[string]interface{}
 	for _, cInstance := range d.Get("instances").([]interface{}) {
 		configInstance := cInstance.(map[string]interface{})
 		for _, aInstance := range connector.Instances {
 			if configInstance["id"].(int) == aInstance.Id ||
```

Comment:
We don't need this match check anymore. The list should be maintained and stop looping.